coral/src/app/features/topics/request/TopicRequest.tsx

```diff
@@ -0,0 +1,151 @@
+import { useEffect } from "react";
+import { useMutation } from "@tanstack/react-query";
+import { FieldErrorsImpl, SubmitHandler } from "react-hook-form";
+import { Box, Divider, Flexbox, FlexboxItem } from "@aivenio/aquarium";
+import {
+  Form,
+  SubmitButton,
+  Textarea,
+  TextInput,
+  useForm,
+  NativeSelectKlaw,
+  NativeSelect,
+} from "src/app/components/Form";
+import formSchema, {
+  useExtendedFormValidationAndTriggers,
+} from "src/app/features/topics/request/schemas/topic-request-form";
+import SelectOrNumberInput from "src/app/features/topics/request/components/SelectOrNumberInput";
+import type { Schema } from "src/app/features/topics/request/schemas/topic-request-form";
+import { createMockEnvironmentDTO } from "src/domain/environment/environment-test-helper";
+import { mockGetEnvironmentsForTeam } from "src/domain/environment/environment-api.msw";
+import { useGetEnvironmentsForTeam } from "src/app/features/topics/request/hooks/useGetEnvironmentsForTeam";
+import { Environment } from "src/domain/environment";
+
+const mockedData = [
+  createMockEnvironmentDTO({
+    name: "DEV",
+    id: "2",
+    maxPartitions: undefined,
+    maxReplicationFactor: undefined,
+    defaultPartitions: "2",
+    defaultReplicationFactor: "2",
+    topicprefix: "dev-",
+  }),
+  createMockEnvironmentDTO({
+    name: "TST",
+    id: "1",
+    maxPartitions: "6",
+    maxReplicationFactor: "2",
+    defaultPartitions: "3",
+    defaultReplicationFactor: "2",
+    topicsuffix: "_TEST",
+  }),
+  createMockEnvironmentDTO({
+    name: "PROD",
+    id: "3",
+    maxPartitions: "16",
+    maxReplicationFactor: "3",
+    defaultPartitions: undefined,
+    defaultReplicationFactor: undefined,
+  }),
+];
+
+function TopicRequest() {
+  useEffect(() => {
+    if (window.msw !== undefined) {
+      mockGetEnvironmentsForTeam({
+        mswInstance: window.msw,
+        response: { data: mockedData },
+      });
+    }
+  }, []);
+  const { data: environments } = useGetEnvironmentsForTeam();
+
+  const defaultValues = Array.isArray(environments)
+    ? {
+        environment: undefined,
+        topicpartitions: "2",
+        replicationfactor: "1",
+        topicname: "",
+        remarks: "",
+        description: "",
+      }
+    : undefined;
+
+  const form = useForm<Schema>({
+    schema: formSchema,
+    defaultValues,
+  });
+
+  const { mutate } = useMutation(() => Promise.resolve());
+  const onSubmit: SubmitHandler<Schema> = (data) => {
```

Comment:
This was actually one of my interrogation about this "mock implementation" step in the ACL request form task: what do we do with mutations? I thought we might want to mock them as well, but I guess this step is about implementing the UI structure, and it is not blocked by having no mutation logic.

Error handling and such should probably be done against the actual API and not mocks.

Reply:
I think this kind of implementation keeps the scope tighter. Introducing the real mutation, which executes the API and mocking it, would increase the scope. I like to ship very small increments.